motion_control/drive_controller/include/drive_controller/DifferentialDriveControllerParameters.hpp

```diff
@@ -0,0 +1,120 @@
+                                            max_speed_percentage_(max_speed_percentage),
+                                            loop_period_(loop_period) {};
+
+    /// @brief Set the left motor wheel diameter dimension
+    /// @param wheel_diameter Wheel diameter value (mm)
+    void set_left_wheel_diameter(double wheel_diameter) { left_wheel_diameter_ = wheel_diameter; }
+
+    /// @brief Return left motor wheel diameter value in mm.
+    /// @return double Wheel diameter value (mm)
+    double left_wheel_diameter_mm() const { return left_wheel_diameter_; }
+
+    /// @brief Set the right wheel motor diameter dimension
+    /// @param wheel_diameter Wheel diameter value (mm)
+    void set_right_wheel_diameter(double wheel_diameter) { right_wheel_diameter_ = wheel_diameter; }
+
+    /// @brief Return right motor wheel diameter value in mm.
+    /// @return double Wheel diameter value (mm)
+    double right_wheel_diameter_mm() const { return right_wheel_diameter_; }
+
+    /// @brief Set the left motor constant value.
+    ///        This value needs to be defined by using the motors characteristics from the datasheet.
+    ///        This value is useful to convert linear velocity in mm/s to a speed ration between 0 to 100%.
+    ///        This speed ratio need to by applied to the max motor voltage in order to get the voltage needed
+    ///        to theoretically reach the desired speed.
+    ///
+    /// @note motor_constant = ((60 * Reduction Ratio / velocity Constant (RPM/V)) / Motor Voltage (V)) * 100
+    /// @param motor_constant motor constant (no unit)
+    void set_left_motor_constant(double motor_constant) { left_motor_constant_ = motor_constant; }
+
+    /// @brief Return left motor constant value
+    /// @return double motor constant (no unit)
+    double left_motor_constant() const { return left_motor_constant_; }
+
+    /// @brief Set the right wheel motor constant value.
+    ///        This value needs to be defined by using the motors characteristics from the datasheet.
+    ///        This value is useful to convert linear velocity in mm/s to a speed ration between 0 to 100%.
+    ///        This speed ratio need to by applied to the max motor voltage in order to get the voltage needed
+    ///        to theoretically reach the desired speed.
+    ///
+    /// @note motor_constant = ((60 * Reduction Ratio / velocity Constant (RPM/V)) / Motor Voltage (V)) * 100
+    /// @param motor_constant motor constant (no unit)
+    void set_right_motor_constant(double motor_constant) { right_motor_constant_ = motor_constant; }
+
+    /// @brief Return right motor constant value
+    /// @return double Wheel motor constant (no unit)
+    double right_motor_constant() const { return right_motor_constant_; }
+
+    /// @brief Set the axle track dimension
+    /// @param track_width Track width value (mm)
+    void set_track_width(double track_width) { track_width_ = track_width; }
```

Comment:
Same here, do we need to dynamically change track width ?

Reply:
It will be more probable for the odometer track width size but in case of the encoder mounted on the motor wheel, we can change it according to odometry calibration results too.

Reply:
true, same as previous comment